homeassistant/components/device_tracker/__init__.py

```diff
@@ -486,7 +486,7 @@ def get_vendor_for_mac(self):
             return None
 
         # prevent lookup of invalid macs
-        if not len(self.mac.split(':')) == 6:
+        if not len(self.mac) == 17:
```

Comment:
@w1ll1am23 can you be a little more specific about how it blows up? As I recall I implemented the macvendor lookup to fail gracefully on any error. If it doesn't I will try to fix that.

As for the Macvendor API, it is pretty relaxed in terms of input and accepts MAC addresses in any of the following formats:
- 00-11-22-33-44-55
- 00:11:22:33:44:55
- 00.11.22.33.44.55
- 001122334455
- 0011.2233.4455

https://macvendors.com/api